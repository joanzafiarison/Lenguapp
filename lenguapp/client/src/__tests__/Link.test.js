import React from 'react';
import {render,screen,act} from "@testing-library/react"
//import renderer from 'react-test-renderer';
// not available for react 17?
import Link from '../Components/Link.js';

it('changes the class when hovered', () => {

  act(()=>{
    render(
        <Link page="http://www.facebook.com">Facebook</Link>
        );
    const linkText = screen.getByText(/facebook/i);
    expect(linkText.textContent).toBe("Facebook");
  })

  act(()=>{
    render(
        <Link page="http://www.instagram.com">Instagram</Link>
        );
    const linkText = screen.getByText(/instagram/i);
    expect(linkText.textContent).toBe("Instagram");
  })


  /* Comparaison de snapshot ecrit => dans __snapshots__ / structure du component
  
    let tree = renderer.create(
      <Link page="https://www.tiktok.com/">Tik Tok</Link>
      );
      expect(tree.toJSON()).toMatchSnapshot();


  
    console.log("tree creation",tree.toJSON())

    renderer.act(()=>{
        tree.toJSON().props.onMouseEnter();
    })
    
    console.log("tree onmouseEnter",tree.toJSON())
    // re-rendering
    
    expect(tree.toJSON()).toMatchSnapshot();

  // manually trigger the callback

    renderer.act(()=>{
        tree.toJSON().props.onMouseLeave();
    })
    
    console.log("tree onmouseLeave",tree.toJSON())
    // re-rendering
    
    expect(tree.toJSON()).toMatchSnapshot();
  */

});