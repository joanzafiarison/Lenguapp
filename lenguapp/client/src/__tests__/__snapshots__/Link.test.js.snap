// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<a
  className="normal"
  href="https://www.tiktok.com/"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Tik Tok
</a>
`;

exports[`changes the class when hovered 2`] = `
<a
  className="hovered"
  href="https://www.tiktok.com/"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Tik Tok
</a>
`;

exports[`changes the class when hovered 3`] = `
<a
  className="normal"
  href="https://www.tiktok.com/"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Tik Tok
</a>
`;
